userdocs/diagnostics/branches-have-mismatching-types.md

```diff
@@ -0,0 +1,35 @@
+# Branches have mismatching types
+
+When an `if` or `switch` statement is used as an expression, each of its branches must produce a value of the same type when type checked independently.
```

Comment:
The first paragraph of these articles is used as the DocC summary. I think this is a fine summary, but it's strange for the article itself to start with "For example," in the next paragraph. For the ones I've written, I write one brief sentence explaining the issue and one brief sentence about resolving it, here's an example

> Sharing mutable state between concurrent tasks can cause data races in your program. Resolve this error by only accessing mutable state in one task at a time.

Maybe something like this?

> Result values in `if` and `switch` expressions share the same type. Resolve this error by providing a type annotation or adjusting the values to have matching types.